prov/shm/src/smr_ep.c

```diff
@@ -247,6 +247,33 @@ static uint64_t smr_tx_comp_flags(uint32_t op)
 	return flags;
 }
 
+static uint64_t smr_rx_comp_flags(uint32_t op, uint16_t op_flags)
+{
+	uint64_t flags = 0;
+
+	switch (op) {
+	case ofi_op_tagged:
+		flags = FI_TAGGED;
+		/* fall through */
+	case ofi_op_msg:
+		flags |= FI_RECV;
+		break;
+	case ofi_op_write:
+		flags = FI_RMA | FI_REMOTE_WRITE;
+		break;
+	case ofi_op_read_req:
+		flags = FI_RMA | FI_REMOTE_READ;
+		break;
+	default:
+		break;
+	}
+
+	if (op_flags & OFI_REMOTE_CQ_DATA)
+		flags |= FI_REMOTE_CQ_DATA;
+
+	return flags;
+}
```

Comment:
Rather than using a switch statement, can this be converted into an array?  E.g. flags = op_flags[op], where the opcode is the index into the array.